### PR TITLE
Implement markBootSuccessfulAllSnaps() and add test

### PR DESCRIPTION
**Pull request by mvo5**
Small branch that implements the boot-success logic for all-snaps.

It will simply store the current good os and kernel version in the bootloader variables snappy_os_good and snappy_kernel_good if it detects that its running on an all-snap system. The test simulates such a system by mocking the lsblk output.

### CODE REVIEW

**Comment by zyga** [comment]
Is there any place that contains documentation of all the bootloader variables that also needs to be changed?

File: partition/partition.go [hunk 1]
```diff
@@ -352,9 +352,38 @@ func (p *Partition) markBootSuccessfulSnappyAB() error {
...
 
-// FIXME: stub
 func (p *Partition) markBootSuccessfulAllSnaps() error {
-	panic("markBootSuccessfulAllSnaps is not implemented yet")
+	bootloader, err := bootloader(p)
+	if err != nil {
+		return err
+	}
+
+	// FIXME: we should have something more atomic here, i.e. one write
```
Comment (zyga) [inline partition/partition.go]: I second this but unless there's an API for doing that let's not block this

**Comment by zyga** [comment]
+1 apart for the comments above

**Comment by chipaca** [comment]
LGTM; :+1:

About atomicity and doing things in a single write, not those are separate things ;-)
Three issues: you're doing read-modify-write-write without holding a lock, so it is in theory possible for bad things to happen (in practice i'm not worried, because AIUI it isn't possible to be running this code twice from different places or sth), you're doing two writes (which is I think what you meant), and the write isn't atomic in any sense (which is AIUI not something we can fix).

I'm still fine to not block on this, but thought I'd mention all the bad :-)

**Comment by mvo5** [comment]
Thanks @zyga for the comment about the documentation. There is indeed doc/system-updates.rst that will need updating once the new mechanism is fully done.

**Comment by mvo5** [comment]
@chipaca meh, I really need to improve my language skills. I changed the comment to take your suggestions into account.

**Pull request merged**